addons/mail/controllers/discuss.py

```diff
@@ -98,6 +98,8 @@ def _response_discuss_channel_invitation(self, channel_sudo, is_channel_token_se
                         'shouldAddGuestAsMemberOnJoin': True,
                         'shouldDisplayWelcomeViewInitially': True,
                     })
+                if not channel_sudo._can_invite_guest(guests=guest):
+                    raise NotFound()
```

Comment:
again, I would move the check above the if/else block

Reply:
Same here if we want guests in the check of method `_can_invite_guest`, we need to put this after the creation of guests.